webapi/Controllers/ServiceInfoController.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Diagnostics;
+using System.Reflection;
+using CopilotChat.WebApi.Models.Response;
+using CopilotChat.WebApi.Options;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.SemanticMemory;
+
+namespace CopilotChat.WebApi.Controllers;
+
+/// <summary>
+/// Controller responsible for returning information on the service.
+/// </summary>
+[ApiController]
+public class ServiceInfoController : ControllerBase
+{
+    private readonly ILogger<ServiceInfoController> _logger;
+
+    private readonly IConfiguration Configuration;
+
+    private readonly SemanticMemoryConfig memoryOptions;
+    private readonly ChatAuthenticationOptions _chatAuthenticationOptions;
+    private readonly FrontendOptions _frontendOptions;
+
+    public ServiceInfoController(
+        ILogger<ServiceInfoController> logger,
+        IConfiguration configuration,
+        IOptions<SemanticMemoryConfig> memoryOptions,
+        IOptions<ChatAuthenticationOptions> chatAuthenticationOptions,
+        IOptions<FrontendOptions> frontendOptions)
+    {
+        this._logger = logger;
+        this.Configuration = configuration;
+        this.memoryOptions = memoryOptions.Value;
+        this._chatAuthenticationOptions = chatAuthenticationOptions.Value;
+        this._frontendOptions = frontendOptions.Value;
+    }
+
+    /// <summary>
+    /// Return the memory store type that is configured.
+    /// </summary>
+    [Route("serviceOptions")]
+    [HttpGet]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public IActionResult GetServiceOptions()
+    {
+        var response = new ServiceOptionsResponse()
+        {
+            MemoryStore = new MemoryStoreOptionResponse()
+            {
+                Types = Enum.GetNames(typeof(MemoryStoreType)),
+                SelectedType = this.memoryOptions.GetMemoryStoreType(this.Configuration).ToString(),
+            },
+            Version = GetAssemblyFileVersion()
+        };
+
+        return this.Ok(response);
+    }
+
+    /// <summary>
+    /// Return the settings to be used by the frontend client to this service.
+    /// </summary>
+    [Route("frontendConfig")]
```

Comment:
The way I see it, this is extensible to more than just auth. For example, we could have feature flags and whatnot in there so we can change the behavior of the frontend only by changing our settings instead of making a new deployment. That's why I don't want to narrow the name down to only auth,

Given that, would you rather have /config, or /settings?

Reply:
> The way I see it, this is extensible to more than just auth. For example, we could have feature flags and whatnot in there so we can change the behavior of the frontend only by changing our settings instead of making a new deployment.

this was the intention of the `/serviceOptions` route above this one. since your new route includes `AllowAnonymous`, I think we need to be careful about the info that we expose here and to that point should limit the scope to the bare minimum auth info needed to get the app running. everything else should go through the auth flow.

Reply:
Sold